odl/discr/partition.py

```diff
@@ -451,7 +451,14 @@ def __eq__(self, other):
             return True
 
         # Optimized version for exact equality
-        return self.set == other.set and self.grid == other.grid
+        return (isinstance(other, type(self)) and
```

Comment:
This was an old bug where we did not properly test that other had the required properties, which gave an exception. We didn't find ti since we lacked tests (we now do).

Anyway, can someone remind me why we use the pattern:

```
isinstance(other, type(self)) and isinstance(self, type(other))
```

instead of the much more simple

```
type(self) == type(other)
```